Formula/g/go-librespot.rb

```diff
@@ -0,0 +1,61 @@
+class GoLibrespot < Formula
+  desc "Yet another open source Spotify client, written in Go"
+  homepage "https://github.com/devgianlu/go-librespot"
+  url "https://github.com/devgianlu/go-librespot/archive/refs/tags/v0.3.3.tar.gz"
+  sha256 "15678b4d6954b381fb9c5a84aca2b0d7706e4d0b2c3644bafcb79cea6d31064c"
+  license "GPL-3.0-only"
+
+  head "https://github.com/devgianlu/go-librespot.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+  depends_on "pkgconf" => :build
+  depends_on "libogg"
+  depends_on "libvorbis"
+  on_linux do
+    depends_on "alsa-lib"
+  end
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/devgianlu/go-librespot.version=#{version}
+    ]
+    ldflags << "-X github.com/devgianlu/go-librespot.commit=#{Utils.git_short_head(length: 8)}" if build.head?
+
+    system "go", "build", *std_go_args(output: bin/"go-librespot", ldflags: ldflags), "./cmd/daemon"
+  end
+
+  def post_install
+    (var/"log").mkpath
+    cfg_dir = etc/"go-librespot"
+    cfg_dir.mkpath
+    # On macOS, create a minimal config that selects the correct backend.
+    if OS.mac?
+      cfg_file = cfg_dir/"config.yml"
+      unless cfg_file.exist?
+        cfg_file.write <<~YAML
+          # Auto-generated by Homebrew. You can edit this file.
+          # Use CoreAudio via AudioToolbox on macOS.
+          audio_backend: audio-toolbox
+        YAML
+      end
+    end
+  end
+
+  service do
+    run [opt_bin/"go-librespot", "--config_dir", etc/"go-librespot"]
+    keep_alive true
+    log_path var/"log/go-librespot.log"
+    error_log_path var/"log/go-librespot.log"
+  end
+
+  test do
+    help = shell_output("#{bin}/go-librespot -h")
+    assert_match "the configuration directory", help
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
Thanks for the link! I now feed `go-librespot` an invalid configuration and assert on the error.